source/updateCheck.py

```diff
@@ -331,20 +331,22 @@ class UpdateResultDialog(
 ):
 	helpId = "GeneralSettingsCheckForUpdates"
 
-	def __init__(self, parent, updateInfo, auto):
+	def __init__(self, parent, updateInfo: Optional[Dict], auto: bool) -> None:
 		# Translators: The title of the dialog informing the user about an NVDA update.
 		super().__init__(parent, title=_("NVDA Update"))
 
 		self.updateInfo = updateInfo
 		mainSizer = wx.BoxSizer(wx.VERTICAL)
 		sHelper = guiHelper.BoxSizerHelper(self, orientation=wx.VERTICAL)
 
+		remoteUpdateExists = updateInfo is not None
 		pendingUpdateDetails = getPendingUpdate()
-		canOfferPendingUpdate = isPendingUpdate() and pendingUpdateDetails[1] == updateInfo["version"]
+		remoteUpdateMatchesPending = pendingUpdateDetails[1] == updateInfo["version"]

```

Comment:
Shouldn't this line check remoteUpdateExists as well? Otherwise is rupdateInfo is None there will be a KeyError.

Reply:
oops, my linting fix reintroduced the problem